expyfun/_experiment_controller.py

```diff
@@ -80,6 +80,20 @@ class ExperimentController:
         The desired dB SPL at which to play the stimuli.
     noise_db : float
         The desired dB SPL at which to play the dichotic noise.
+    noise_dur : float
+        The minimum duration of the noise in seconds. It will be rounded up so
+        that the length of the noise at stim_fs is a power of 2 (required
+        for the ringbuffer). Default is 15 seconds.
+    noise_array : ndarray | None
+        An array containing the noise to be presented. Must have a length that
+        is a power of 2, and generated with a reference RMS amplitude of 1 for
+        proper scaling. It is assumed that the sample rate matches stim_fs.
+        If None, additive white Gaussian noise will be generated.
+    check_noise_rms : bool | str
+        Whether or not to check the RMS of noise_array. Recommended to be True
+        (default) to ensure scaling works properly. Can also be 'max', in
+        which case it is only checked that the RMS is less than or equal to
+        the reference RMS of 1.
```

Comment:
Do we need all of these? It would be nice to make as few API additions as possible to solve your use case. How about just:
```
noise_array : ndarray | None
    An array ... .
    If not None, RMS is assumed to be 1, and this will not be checked.
    The amplitude will be scaled to achieve the desired ``noise_db`` under the assumption
    (which will not be checked) that the RMS of the input array is 1.
```
This would suppress RMS checking entirely and allow setting arbitrary durations as well. So it should work for your use case I think. Basically it's like an advanced option so disabling checks etc. makes sense I think

Reply:
fair enough. I've removed those options so it's just `noise_array` now.